src/java.base/share/classes/java/util/HashSet.java

```diff
@@ -372,4 +380,21 @@ public Object[] toArray() {
     public <T> T[] toArray(T[] a) {
         return map.keysToArray(map.prepareArray(a));
     }
+
+    /**
+     * Creates a new, empty HashSet suitable for the expected number of items.
+     * The returned set uses the default load factor of 0.75, and its initial capacity is
+     * generally large enough so that the expected number of items can be added
+     * without resizing the set.
+     *
+     * @param numItems    the expected number of items
+     * @param <E>         the type of keys maintained by this set
+     * @return the newly created set
+     * @throws IllegalArgumentException if numItems is negative
+     * @since 19
+     */
+    public static <E> HashSet<E> newHashSet(int numItems) {
+        return new HashSet<>(HashMap.calculateHashMapCapacity(numItems));
+    }
```

Comment:
Please use "elements" instead of "items" throughout for the objects that are members of the HashSet. This includes the API notes above as well as the specs and API notes in LinkedHashSet.

Reply:
@stuart-marks done.